### PR TITLE
[#17744] - Pressing clear button of calendar on required date field submits admin form in Joomla! 3.7.5 --- corrected

### PR DESCRIPTION
**Pull request by akritianand**
Pull Request for Issue # 17744

### Summary of Changes
Previously, when we edit an article under publishing tab, the clear button in calendar field did not work if it was a required field.
Now, it clears the calendar field when pressed instead of redirecting (as it did originally).
Changes were made in calendar.js and calendar.php

### Testing Instructions
Create an admin form with a required date field. i.e. open administrator\components\com_content\models\forms\article.xml and make field "publish_down" required by adding required="true" to field's xml node.
Go to Joomla! administration panel and open an article edit view.
Finish Publishing field should now be a required field.
Add Finish Publishing date, using the date picker of the calendar
Click "Clear" button on the Calendar.

### CODE REVIEW

File: media/system/js/fields/calendar.js [hunk 1]
```diff
@@ -784,11 +784,11 @@
 		row = createElement("div", this.wrapper);
 		row.className = "buttons-wrapper btn-group";
 
-		this._nav_save = hh(JoomlaCalLocale.save, '', 100, 'button', '', 'js-btn btn btn-clear', {"data-action": "clear"});
+		this._nav_clear = hh(JoomlaCalLocale.clear, '', 100, 'button', '', 'js-btn btn btn-clear', {"type": "button", "data-action": "clear"});
 
-		if (!this.inputField.hasAttribute('required')) {
-			var savea = row.querySelector('[data-action="clear"]');
-			savea.addEventListener("click", function (e) {
+		if (1) {
```
Comment (izharaazmi) [inline media/system/js/fields/calendar.js]: Why we don't need to check for the `required` attribute? And why `if (1)`?

**Comment by izharaazmi** [comment]
@akritianand You need to generate the `calendar.min.js` too.

File: media/system/js/fields/calendar.js (hunk 1 above)
Comment (izharaazmi) [reply media/system/js/fields/calendar.js]: @akritianand IMHO, the clear button should be hidden when field is `required`.

File: layouts/joomla/form/field/calendar.php [hunk 2]
```diff
@@ -100,7 +100,7 @@
 // The static assets for the calendar
 JHtml::_('script', $localesPath, false, true, false, false, true);
 JHtml::_('script', $helperPath, false, true, false, false, true);
-JHtml::_('script', 'system/fields/calendar.min.js', false, true, false, false, true);
+JHtml::_('script', 'system/fields/calendar.js', false, true, false, false, true);
```
Comment (brianteeman) [inline layouts/joomla/form/field/calendar.php]: please revert this change and update the calendar.min.js file with your changes

**Comment by mbabker** [comment]
Also needs updating to deconflict with the now merged https://github.com/joomla/joomla-cms/pull/17777

File: media/system/js/fields/calendar.js (hunk 1 above)
Comment (akritianand) [reply media/system/js/fields/calendar.js]: Okay, if that is required, then there was not any problem initially. Because the clear button still showed up if the field was required; it just did not perform its function.

**Comment by ghost** [comment]
similar #17809<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17793">issues.joomla.org/tracker/joomla-cms/17793</a>.</sub>

**Comment by izharaazmi** [comment]
This pr is broken and we have #17809 for the same issue. This should be closed, imho.

**Comment by ghost** [comment]
closed as stated above.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17793">issues.joomla.org/tracker/joomla-cms/17793</a>.</sub>
